<launch>
  <rosparam command="load" file="$(find smabo)/rcalib.yaml" />
  <rosparam command="load" file="$(find smabo)/config/vt.yaml" />
  <node pkg="rovi_utils" type="cropper.py" name="surface_cropper" output="screen">
    <remap from="~config" to="/config/cropper" />
    <remap from="~param" to="/cropper" />
    <remap from="~update" to="/update/cropper" />
    <remap from="~capture" to="/request/capture" />
    <remap from="~captured" to="/response/capture" />
    <remap from="~clear" to="/request/clear" />
    <remap from="~redraw" to="/request/redraw" />
    <remap from="~in/floats" to="/sensors/floats" />
    <remap from="~out/floats" to="/crop/floats" />
    <remap from="~raw/floats" to="/crop/raw/floats" />
  </node>
  <node pkg="smabo" type="searcher.py" name="searcher" args="path:=$(find smabo)/recipe" output="screen">
    <remap from="~config" to="/config/searcher" />
    <remap from="~param" to="/searcher" />
    <remap from="~in/surface/floats" to="/crop/floats" />
    <remap from="~master/surface/floats" to="/master/surface/floats" />
    <remap from="~clear" to="/request/clear" />
    <remap from="~load" to="/request/model_load" />
    <remap from="~loaded" to="/response/recipe_load" />
    <remap from="~save" to="/request/model_save" />
    <remap from="~saved" to="/response/model_save" />
    <remap from="~solve" to="/request/solve" />
    <remap from="~solved" to="/searcher/solved" />
    <remap from="~redraw" to="/request/redraw" />
    <remap from="~score" to="/searcher/score" />
    <remap from="~stat" to="/searcher/stat" />
  </node>
  <node pkg="smabo" type="picker.py" name="" output="screen">
    <remap from="~config" to="/config/picker" />
    <remap from="~param" to="/picker" />
    <remap from="~clear" to="/request/clear" />
    <remap from="~solve" to="/searcher/solved" />
    <remap from="~score" to="/searcher/score" />
    <remap from="~solved" to="/response/solve" />
    <remap from="~redraw" to="/request/redraw" />
  </node>
</launch>
